src/Microsoft.Health.Fhir.Liquid.Converter/Filters/SegmentFilters.cs

```diff
@@ -128,5 +128,50 @@ public static bool HasSegments(Hl7v2Data hl7v2Data, string segmentIdContent)
 
             return result;
         }
+
+        public static List<Hl7v2Data> SliceDataBySegments(Hl7v2Data hl7v2Data, List<Hl7v2Segment> segmentList, Hl7v2Segment endSegment = null)
+        {
+            var results = new List<Hl7v2Data>();
+            var result = new Hl7v2Data();
+            string[] segmentSeparators = { "\r\n", "\r", "\n" };
+            var valueList = hl7v2Data.Value.Split(segmentSeparators, StringSplitOptions.RemoveEmptyEntries);
```

Comment:
I think splitting the input data into segments is a common logic, can we extract it into separated function?

Reply:
We can add an utility in engine to do so and also replace the same logics in [Hl7v2DataParser](https://github.com/microsoft/FHIR-Converter/blob/ae582fa986a74def8e1a5736162c62571a315d9e/src/Microsoft.Health.Fhir.Liquid.Converter/Parsers/Hl7v2DataParser.cs#L20).

Reply:
Yes, I think so.